src/site/content/en/learn/css/spacing/index.md

```diff
@@ -0,0 +1,380 @@
+These are applied as follows:
+
+- One value will be applied to all sides. (i.e. `margin: 20px`)
+- Two values: the first value will be applied to the top and bottom sides,
+and the second value will be applied to the left and right sides.
+(i.e. `margin: 20px 40px`)
+- Three values: the first value is `top`,
+the second value is `left` **and** `right`,
+and the third value is `bottom`. (i.e. margin: 20px 40px 30px`)
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'abpwqyQ'
+} %}
+
+Margin can be defined with a length,
+percentage or auto value, such as `1em` or `20%`.
+If you use a percentage,
+the value will be calculated based on the width of your element's containing block.
+
+This means that if your element's containing block has a width of `250px`
+and your element has a `margin` value of `20%`:
+each side of your element will have a computed margin of `50px`.
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'YzNQeyb'
+} %}
+
+You can also use a value of `auto` for margin.
+For block level elements with a restricted size,
+an `auto` margin will take up available space in the direction that it is applied to.
+A good example is this one,
+from the [flexbox module](/learn/css/flexbox), where the items push away from each other.
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'poRELbR'
+} %}
+
+Another good example of `auto` margin is a horizontally centered wrapper which has a max width.
+This sort of wrapper is often used to create a consistent center column on a website.
+
+```css
+.wrapper {
+	max-width: 400px;
+	margin: 0 auto;
+}
+```
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'XWpgZKq'
+} %}
+
+Here, margin is removed from the top and bottom (block) sides,
+and `auto` shares the space between the left and right (inline) sides.
+
+{% Aside %}
+In the last lesson on [logical properties](/learn/css/logical-properties),
+you learned that instead of specifying `margin-top`, `margin-right`, `margin-bottom` and `margin-left`,
+you can use `margin-block-start`, `margin-inline-end`, `margin-block-end` and `margin-inline-start`.
+{% endAside %}
+
+### Negative margin
+
+Negative values can also be used for margin.
+Instead of adding space between adjacent sibling elements,
+it will **reduce space** between them.
+This can result in overlapping elements,
+if you declare a negative value that's more than the available space.
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'YzNQeLP'
+} %}
+
+### Margin collapse
+
+Margin collapse is a tricky concept,
+but it's something you'll run into very commonly when building interfaces.
+Say you have two elements: a heading and a paragraph that both have vertical margin on them:
+
+```html
+<article>
+  <h1>My heading with teal margin</h1>
+  <p>A paragraph of text that has blue margin on it, following the heading with margin.</p>
+</article>
+```
+
+```css
+h1 {
+	margin-bottom: 2rem;
+}
+
+p {
+	margin-top: 3rem;
+}
+```
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'OJWgvMx'
```

Comment:
An idea for this demo:

Let the user change the `margin-bottom` on the h1 with a slider to directly see how margin collapse works.

Reply:
Good idea: sorted